sdk/python/kfp/dsl/_pipeline_volume.py

```diff
@@ -69,9 +72,11 @@ def __init__(self,
                 init_volume["persistent_volume_claim"] = pvc_volume_source
         super().__init__(**init_volume, **kwargs)
         if not name_provided:
-            self.name = "pvolume-%s" % hashlib.sha256(
-                bytes(json.dumps(self.to_dict(), sort_keys=True), "utf-8")
-            ).hexdigest()
+            hash_value = hashlib.sha256(bytes(json.dumps(self.to_dict(),
+                                                         sort_keys=True),
+                                              "utf-8")).hexdigest()
+            name = "pvolume-{}".format(hash_value)
+            self.name = name[0:63] if len(name) > 63 else name
```

Comment:
JFYI: `name[0:63]` works fine even when length is less than 63 characters.

Reply:
Oh, didn't know, thanks! Should I change and push it?